File: backend/utils/sanitizer.py
import re

def sanitize_string(value, max_length=None):
    """Sanitize string input by stripping whitespace, control characters, and limiting length."""
    if value is None:
        return ""
    if not isinstance(value, str):
        value = str(value)
    
    # Remove null bytes and control characters
    sanitized = re.sub(r'[\x00-\x1f\x7f-\x9f]', '', value)
    sanitized = sanitized.strip()
    
    if max_length and len(sanitized) > max_length:
        sanitized = sanitized[:max_length]
    
    return sanitized

File: lib/sanitizer.js
/**
 * Sanitize string input by stripping whitespace, control characters, and limiting length.
 * Direct port of backend/utils/sanitizer.py
 */
export function sanitizeString(value, maxLength = null) {
  if (value === null || value === undefined) {
    return "";
  }

  let sanitized = String(value);

  // Remove null bytes and control characters (matching Python: [\x00-\x1f\x7f-\x9f])
  sanitized = sanitized.replace(/[\x00-\x1f\x7f-\x9f]/g, "");
  sanitized = sanitized.trim();

  if (maxLength && sanitized.length > maxLength) {
    sanitized = sanitized.substring(0, maxLength);
  }

  return sanitized;
}
